### PR TITLE
Development Into Master

### PR DESCRIPTION
**Pull request by ImSighted**

### CODE REVIEW

**Review by Maxine770 — COMMENTED** [review]
Met the minimum requirements, but could make dice rolling continous and report individual dice rolls.
Several naming violations per the C# google style guide, consider correcting.
Should implement named constants instead of magic numbers.
Certain properties are exposed when they don't have to be, like the random object in the die class, consider making these private.
Code is readable and comments added when necessary, perhaps excessively.
Errors handled correctly and test cases are rigorous but error messages could be more descriptive for example by including the offending value and the expected range.

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -14,10 +14,16 @@ internal class Die
          */
 
         //Property
-
-
+        public int die_num { get; set; } // establishes get and set for die_num
+        static Random rnd = new Random(); // makes a new random object for the class, outside 'Roll()' method to allow for different random per roll
```
Comment (Maxine770) [inline CMP1903_A1_2324/Die.cs]: Random instance can be made read-only and private.

File: CMP1903_A1_2324/Die.cs [hunk 2]
```diff
@@ -14,10 +14,16 @@ internal class Die
          */
 
         //Property
-
-
+        public int die_num { get; set; } // establishes get and set for die_num
```
Comment (LucyferYX) [inline CMP1903_A1_2324/Die.cs]: Formatting: `die_num` is a public property and should use PascalCase.  I suggest changing to something like `DieNum`.

File: CMP1903_A1_2324/Testing.cs [hunk 3]
```diff
@@ -16,5 +16,21 @@ internal class Testing
          */
 
         //Method
+
+        public void Check_Dice_Number()
```
Comment (LucyferYX) [inline CMP1903_A1_2324/Testing.cs]: Formatting: Method name is PascalCase but should not use underscores. I suggest changing it to something like `CheckDiceNumber()`.

**Comment by LucyferYX** [comment]
**Requirements**
The code fulfils basic requirements of performing dice rolls and reporting sum of three rolls.
Extra requirements could be met with making dice rolls continuous and reporting individual rolls.

**Formatting**
The code isn’t entirely formatted according to the C# style guidelines.
Many properties, methods and variables use underscores in their naming which is not in line with the C# style guidelines.
Public properties should use PascalCase, but private ones should use camelCase with underscore before them.
For example, `die_num` is a public property and should use PascalCase.

**Readability**
The code is readable, but could use some adjustments.
I suggest using named constants when testing die roll range.
Nearly every single code line has a comment, which doesn't help the code readability.
I suggest deducing some comments that simply restate what the code is doing.
For example, `return die_total;` does not need a comment `// returns die_total` because it's self-explanatory.

**Error handling**
The errors are handled correctly, but could explain the issues better.
For example, when testing the roll type, the error message could include the actual type and the expected type for die roll.

**Pull request merged**